remote/mdb_util_test.go

```diff
@@ -12,30 +13,101 @@ import (
 )
 
 func makeTestMDBServiceAndClient(ctx context.Context, mngr jasper.Manager) (Manager, error) {
-	addr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("localhost:%d", testutil.GetPortNumber()))
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
+	var (
+		addr   net.Addr
+		client Manager
+		err    error
+	)
+tryPort:
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, errors.Wrap(ctx.Err(), "starting MDB service")
+		default:
+			addr, err = net.ResolveTCPAddr("tcp", fmt.Sprintf("localhost:%d", testutil.GetPortNumber()))
+			if err != nil {
+				continue tryPort
+			}
 
-	closeService, err := StartMDBService(ctx, mngr, addr)
-	if err != nil {
-		return nil, errors.WithStack(err)
-	}
-	go func() {
-		<-ctx.Done()
-		grip.Notice(closeService())
-	}()
-	if err = testutil.WaitForWireService(ctx, addr); err != nil {
-		return nil, errors.WithStack(err)
-	}
+			srvCtx, srvCancel := context.WithCancel(ctx)
+			closeService, err := StartMDBService(srvCtx, mngr, addr)
+			if err != nil {
+				srvCancel()
+				return nil, errors.WithStack(err)
+			}
+
+			failedToConnect := make(chan struct{})
+			go func() {
+				defer func() {
```

Comment:
the same question about combining the goroutines

Reply:
This already has just one goroutine, so I didn't change anything.